### PR TITLE
Consider RANs in variable ordering

### PR DESCRIPTION
**Pull request by nafur**
The variable ordering in the arithmetic normal form should also consider real algebraic numbers. This PR also adds new comparators (essentially doing the same) in the rewriter that will replace the normal form utility classes.

### CODE REVIEW

File: src/theory/arith/arith_rewriter.cpp [hunk 1]
```diff
@@ -42,6 +42,88 @@ namespace arith {
...
+    if (aIsVar != bIsVar) return aIsVar;
+
+    return a < b;
+  }
+};
+
+/**
+ * Implements an ordering on arithmetic nonlinear multiplications. As we assume
+ * rationals to be handled separately, we only consider Kind::NONLINEAR_MULT as
+ * multiplication term. For individual factors of the product, we rely on the
```
Comment (4tXJ7f) [inline src/theory/arith/arith_rewriter.cpp:84]:
```suggestion
 * multiplication terms. For individual factors of the product, we rely on the
```

File: src/theory/arith/arith_rewriter.cpp [hunk 2]
```diff
@@ -42,6 +42,88 @@ namespace arith {
...
+    if (a == b) return false;
+
+    Assert(a.getKind() != Kind::MULT);
+    Assert(b.getKind() != Kind::MULT);
+
+    bool aIsMult = a.getKind() == Kind::NONLINEAR_MULT;
+    bool bIsMult = b.getKind() == Kind::NONLINEAR_MULT;
+    if (aIsMult != bIsMult) return !aIsMult;
+
+    if (!aIsMult)
```
Comment (4tXJ7f) [inline src/theory/arith/arith_rewriter.cpp:105]: Shouldn't this be:

```suggestion
    if (!aIsMult && !bIsMult)
```

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
Looks mostly good to me, but I have a question.

File: src/theory/arith/arith_rewriter.cpp (hunk 2 above)
Comment (nafur) [reply src/theory/arith/arith_rewriter.cpp:105]: As this point, we know that `aIsMult == bIsMult`, so it's sufficient to only check one.

**Review by ajreynol — APPROVED** [review]

File: src/theory/arith/arith_rewriter.cpp (hunk 2 above)
Comment (4tXJ7f) [reply src/theory/arith/arith_rewriter.cpp:105]: Ah, fair enough.

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the explanation.

**Pull request merged**